README.md

```diff
@@ -537,6 +537,30 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
+* <a id='destructure-inline-let'></a> (<a href='#destructure-inline-let'>link</a>) **When destructuring an enum case or a tuple, place the `let` keyword inline, adjacent to each individual property assignment.** [![SwiftFormat: hoistPatternLet](https://img.shields.io/badge/SwiftFormat-hoistPatternLet-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#hoistPatternLet)
```

Comment:
Putting here so we have a thread:

>  2. For consistency, we should prefer either _always_ inlining the `let` keywords, or _never_ inlining the keywords. Since enum payloads with a small number of associated values are much more common than enum payloads with a large number of associated values, we should prefer always inlining the `let` keywords.

For an enum with one associated value, the number of characters is the same in either style. For an enum with multiple associated values, the proposed style is more verbose. I don't follow how this proposed style prioritizes enum cases with fewer associated values.

Reply:
In our codebase, when destructuring an enum case with a single associated value, inline let (`case .success(let string)`) is an order of magnitude more common than hoisted let (`case let .success(string)`).

| | Example | Regex | Count |
| --- | --- | ---- | --- |
| Inline let | `case .success(let string)` | `case \.[A-Za-z]+\(let [^,]*\):` | 6842 |
| Hoisted let | `case let .success(string)` | `case let \.[A-Za-z]+\([^,]*\):` | 901 |

Looking generally at the use of inline vs hoisted let overall (with any number of arguments), you can see that inline is overall 4x more common that hoisted let:

| | Example | Regex | Count |
| --- | --- | ---- | --- |
| Inline let | `case .success(` | `case \.[A-Za-z]+\(` | 9404 |
| Hoisted let | `case let .success(` | `case let \.[A-Za-z]+\(` | 2395 |

Because of this clear existing preference, if we are going to pick between the two options, I think we should choose to inline the `let` keywords.

Reply:
This is very helpful.

Reply:
IIRC inline let is the Xcode default (we should verify)? I think it would be nice to include some high-level reasoning in the rule (e.g. "Xcode autocomplete pushes you this direction and we've seen this in practice in a large code base").

Reply:
I would also be interested to know how many of our case have multiple associated values where we could save characters by moving from the more common inline let to the hoisted let.

Reply:
According to the numbers above, we have:
 - 7,743 cases with a single associated value (the sum of the two rows in the first table)
 - 4,056 cases with more than one associated value (the sum of the rows in the second table, subtracting the cases with a single associated value)

I would personally favor consistency (always placing the `let` keyword in the same place) over some sort of conditional logic based on the number of associated value cases.

Reply:
> I would personally favor consistency (always placing the `let` keyword in the same place) over some sort of conditional logic based on the number of associated value cases.

Yes I am in full agreement here. We should have a single consistent rule.

Reply:
Sorry for not piecing that data together myself. It makes sense that people inline `let` less for cases with multiple associated values, since the hoisted `let` saves characters when there are multiple associated values.

Reply:
I can get onboard with preferring inline let in call cases though I think it would be good to make the reasoning stronger in the rule.

Reply:
Updated the reasoning in the rule and PR description.

Reply:
@calda I'm not seeing reasoning in the rule itself. Do you need to push a commit?
![Screen Shot 2021-05-14 at 4 22 02 PM](https://user-images.githubusercontent.com/1791049/118340454-94f1bd80-b4d0-11eb-90a4-dd607881ffb0.png)

Reply:
I made the changes locally but forgot to push them -- fixed!